docker-compose-separate-containers.yml

```diff
@@ -1,23 +1,25 @@
-version: '2'
+version: '3'
 services:
   nginx:
     image: nginx
     container_name: nginx
     ports:
       - "80:80"
     volumes:
-      - /etc/nginx/conf.d
+      - nginx_conf:/etc/nginx/conf.d
 
   dockergen:
     image: jwilder/docker-gen
     command: -notify-sighup nginx -watch /etc/docker-gen/templates/nginx.tmpl /etc/nginx/conf.d/default.conf
-    volumes_from:
-      - nginx
     volumes:
       - /var/run/docker.sock:/tmp/docker.sock:ro
       - ./nginx.tmpl:/etc/docker-gen/templates/nginx.tmpl
+      - nginx_conf:/etc/nginx/conf.d
```

Comment:
Docker copies files from the image into the volume when starting the container.

If files are copied from the `dockergen` container into the `nginx-conf` volume, it'd become empty. This would happen when the `dockergen` container is started before the `nginx` container. Adding `nocopy` here ensures that files are never copied from this container.